### DISCUSSION TITLE
[v4] List user custom properties first

### DISCUSSION THREAD

**Discussion by madmoizo** [tailwindlabs/tailwindcss#14972] [Ideas]
Until build tools automatically remove unused custom properties, it would be more convenient to have user-defined custom properties at the top of the list

When you want to change a custom property, you have to click the "show all properties" button and then scroll at the end of the looong list to access your own custom properties.
And you have to do it every time you select a different html element.

<img width="277" alt="image" src="https://github.com/user-attachments/assets/00759ff8-1ee1-4f88-9937-d4639420b568">

**Comment by adamwathan** [comment]
Hey thanks for the feedback! Can you share an example of the sort of custom stuff you're adding that would be helpful to see first?

**Comment by madmoizo** [comment]
Everything inside @theme {} end up at the end of the list.
This way you can override custom properties set by tailwind (should you?)
But devtools behavior hiding custom properties make tweaking in browser really annoying

<img width="749" alt="image" src="https://github.com/user-attachments/assets/0459dd44-a2b3-48c6-b5d6-15a011c2498f">

**Comment by madmoizo** [comment]
Workaround:
```css
@theme {
  --color-blue:;
}
:root {
  --color-blue: blue;
}
`
